roxie/ccd/ccdfile.cpp

```diff
@@ -1418,9 +1427,49 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
         {
             for (;;)
             {
-                toClose.wait(10 * 60 * 1000);  // check expired file handles every 10 minutes 
+                unsigned checkPeriod = 10;  // check expired file handles every 10 minutes, buddyCopying a little more often
+#ifdef _CONTAINERIZED
+                {
+                    CriticalBlock b(crit);
+                    if (buddyCopying.length())
+                        checkPeriod = 1;
+                }
+#endif
+                toClose.wait(checkPeriod * 60 * 1000);
                 if (closing)
                     break;
+#ifdef _CONTAINERIZED
+                if (checkPeriod==1)
+                {
+                    // Periodically recheck the list to see what is now local, and remove them from the buddyCopying list
+                    ICopyArrayOf<ILazyFileIO> checkBuddies;
+                    {
+                        CriticalBlock b(crit);
+                        buddyCopying.swapWith(checkBuddies);
+                        buddyChecking = true;
+                    }
+                    ForEachItemIn(idx, checkBuddies)
+                    {
+                        ILazyFileIO &check = checkBuddies.item(idx);
+                        if (check.isRemote())
+                        {
+                            check.setCopying(false);
+                        }
+                        if (check.isRemote())
```

Comment:
Should have a comment to indicate why it is being rechecked.  (Could be nested within the previous if to make it clearer it isn't a mistake.)

Reply:
Done